nodejs/week2/meal-sharing/src/backend/api/meals-router.js

```diff
@@ -0,0 +1,76 @@
+const express = require("express");
+const req = require("express/lib/request");
+const router = express.Router();
+
+const meals = require("./../data/meals.json");
+
+router.get("/", async (request, response) => {
+  let filteredMeals = meals;
+  const queryParams = request.query;
+
+  if (Object.keys(queryParams).length === 0) {
+    response.send(meals);
+  } else {
+    let checkParam = false;
+    for (key in queryParams) {
+      switch (key) {
+        case "maxPrice":
+          const maxPrice = Number(queryParams[key]);
+          if (!isNaN(maxPrice) && maxPrice) {
+            filteredMeals = filteredMeals.filter(
+              (item) => item.price < maxPrice
+            );
+            checkParam = true;
+          }
+          break;
+        case "limit":
+          const limit = Number(queryParams[key]);
+          if (!isNaN(limit)) {
+            filteredMeals = filteredMeals.slice(0, limit);
+            checkParam = true;
+          }
+          break;
+        case "title":
+          if (queryParams[key]) {
+            filteredMeals = filteredMeals.filter((item) =>
+              item.title.toLowerCase().includes(queryParams[key].toLowerCase())
+            );
+            checkParam = true;
+          }
+          break;
```

Comment:
Can you break down the body of the for loop into smaller functions? It's quite long for a loop.

Reply:
@jslambda  I rewrote this part to avoid using loop.